Quick Editor: More options button in avatar picker

hamorillo opened the pull request:
Closes #411 

### Description

Part of the milestone `2.1.0` is to add the possibility to modify the alternative text, delete an avatar, or define its rating.

The common entry point for those three options is a button in each avatar in the avatar picker.  

![Image](https://github.com/user-attachments/assets/96a6c095-2bf8-4f33-9727-0d9aec06f6ac)

This PR implements the more options button to allow the parallelized implementation of those three features.

I've added an "Alt Text" button for the sake of define all the architecture and how it should look, however it doesn't execute any action under the hood.

I've generalized our previous `Popup` menu, as this one is pretty similar, so we can keep all the code centralized.

For now, the popup is aligned with `CenterHorizontally`; we can iterate on this if we think that's not the best option.

<img width="250" alt="image" src="https://github.com/user-attachments/assets/cadf625e-439d-441f-ae08-c84b7a9c6797">

### Testing Steps

1. Open the QE in the demo-app
2. Verify you can see the more options button in each of the avatars
3. Click any of the more options buttons and verify the Popup with the menu is opened
4. Smoke test the upload new avatar popup menu

hamorillo committed:
Added more options button in the selectable avatar

hamorillo committed:
Extract PickerPopup from MediaPickerPopup

We need to do a PickerPopup similar to the media one for the avatar options. For that reason, it makes sense to extract the common part to a Composable that can be shared between different Popups.

hamorillo committed:
Add avatar more options popup with AltText option as an example

AdamGrzybkowski commented on gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/components/PickerPopup.kt:
```diff
@@ -100,9 +103,15 @@ private fun PickerPopup(
+                                    text = stringResource(item.text),
+                                    iconRes = item.iconRes,
+                                    contentDescription = stringResource(item.contentDescription),
+                                    shape = RoundedCornerShape(topStart = 8.dp, topEnd = 8.dp),
```
This should be different for the first, middle, and last items. You can see it's incorrect when you open the `MediaPickerPopup` and tap&hold the `Take Photo` button. The ripple will have rounded corners at the top not matching the shape of the button.

hamorillo replied:
Good catch! Updated!

AdamGrzybkowski commented on gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/components/SelectableAvatar.kt:
```diff
@@ -64,10 +80,25 @@ internal fun SelectableAvatar(
+
+        if (moreOptionsPopupVisible) {
+            AvatarMoreOptionsPickerPopup(
+                anchorAlignment = Alignment.CenterHorizontally,
```
It seems like sth is off with the alignment - it's always centered on the screen. Shouldn't it be centered to the anchor? 

It might not be related to this PR, but we might have to work on that calculation to fit this usage a bit better.

hamorillo replied:
I mentioned this in the PR description; I think we can iterate over the alignment in a different PR to keep this one small. But if you think that's a blocker for this one, I can work on it.

AdamGrzybkowski replied:
No, no, not a blocker!

hamorillo replied:
👍 Ticket created -> #419

hamorillo committed:
PopupButton using the proper shape depending in it's position on the list

Depending on the position of the PopupButton and the number of elements in the list, the shape should be different.

AdamGrzybkowski reviewed (approved):
:shipit:
